controlplane/kubeadm/controllers/controller.go

```diff
@@ -245,6 +246,16 @@ func (r *KubeadmControlPlaneReconciler) reconcile(ctx context.Context, cluster *
 	logger := r.Log.WithValues("namespace", kcp.Namespace, "kubeadmControlPlane", kcp.Name, "cluster", cluster.Name)
 	logger.Info("Reconcile KubeadmControlPlane")
 
+	// Check if the Kubernetes Version is supported.
```

Comment:
Is this the right place to put this call?

In case the call is here correct:
Unfortunately this bumped gocyclo to 31, only 30 is allowed. What should I do about that? If the call really belongs here, I think bumping the allowed complexity in the linter is maybe the best option.

Reply:
As discussed. I'll move it one up.